### ISSUE TITLE
Changing the completion value in a web-compatible way

### ISSUE THREAD

**Issue by Jack-Works** [bakkot/do-expressions-v2#10]
Currently the completion value of statements only observable via eval calls. (Is it?)

So if you want to change the completion value for the do expression is very easy. We can make those statements return difference completion value based on the containing evaluation contexts is do expression or eval call.

**Comment by bakkot** [comment]
Yes, I [raised this idea myself a couple years ago](https://github.com/tc39/proposal-do-expressions/issues/21#issuecomment-363521819). At the moment, though, I don't think it's a good path: the only things which have really weird completion values are loops and declarations, and it's simpler to just make the completion values for those forms not observable. Perhaps at a later time we could pursue making them either have different behavior from `eval` or changing `eval` and hoping it's web-breaking (which seems reasonably likely, at least for declarations).
